### PR TITLE
Refactor tests of the image module

### PR DESCRIPTION
**Pull request by rragundez**
### Summary
Refactor tests of the 'image' submodule to match the skeleton of the submodule itself. Now instead of having a single file with all the tests, some logic has been separated into different files according to the class or function being tested. This allows the addition and debugging of failed tests ti be much simpler and faster.

### PR Overview

- [y] This PR requires new unit tests [y/n] (make sure tests are included)
- [n] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [n] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: tests/image/dataframe_iterator_test.py [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+    batch = next(iterator)
+    assert len(batch) == 2
+    assert isinstance(batch[0], np.ndarray)
+    assert isinstance(batch[1], np.ndarray)
+    generator = image_data_generator.ImageDataGenerator()
+    df_iterator = generator.flow_from_dataframe(df, x_col='filepaths')
+    df_iterator_dir = generator.flow_from_dataframe(df, str(tmpdir))
+    df_sparse_iterator = generator.flow_from_dataframe(df, str(tmpdir),
+                                                       class_mode="sparse")
+    if np.isnan(df_sparse_iterator.classes).any():
```
Comment (Dref360) [inline tests/image/dataframe_iterator_test.py]: When does this can happen?

File: tests/image/dataframe_iterator_test.py [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
+    generator = image_data_generator.ImageDataGenerator()
+    df_autoencoder_iterator = generator.flow_from_dataframe(df, str(tmpdir),
+                                                            x_col="filename",
+                                                            y_col=None,
+                                                            class_mode="input")
+
+    batch = next(df_autoencoder_iterator)
+
+    # check if input and output have the same shape and they're the same
+    assert(batch[0].all() == batch[1].all())
```
Comment (Dref360) [inline tests/image/dataframe_iterator_test.py]: The `.all` is misplaced I think.

`np.allclose(batch[0], batch[1])`

File: tests/image/dataframe_iterator_test.py [hunk 3]
```diff
@@ -0,0 +1,653 @@
...
+        assert np.array_equal(a1, a2)
+        assert np.array_equal(a1, a3)
+        assert np.array_equal(c1, c2)
+        assert np.array_equal(c1, c3)
+
+
+def test_dataframe_iterator_n(all_test_images, tmpdir):
+
+    # save the images in the tmpdir
+    count = 0
```
Comment (Dref360) [inline tests/image/dataframe_iterator_test.py]: I'm not sure of the value of this test?

**Comment by rragundez** [comment]
I saw you comments inline @Dref360 , basically I just splitted the current tests that were already there.  I would try to address your questions though.

File: tests/image/dataframe_iterator_test.py (hunk 1 above)
Comment (rragundez) [reply tests/image/dataframe_iterator_test.py]: I don't know. I just moved a test that was already there. But I converted into an assertion

File: tests/image/dataframe_iterator_test.py (hunk 2 above)
Comment (rragundez) [reply tests/image/dataframe_iterator_test.py]: changed

File: tests/image/dataframe_iterator_test.py (hunk 3 above)
Comment (rragundez) [reply tests/image/dataframe_iterator_test.py]: I'm also not sure, it was already there. @Vijayabhaskar96  do you know? I think you originally made  this one.

Comment (Vijayabhaskar96) [reply tests/image/dataframe_iterator_test.py]: Looks like It matches the number of filenames found and the number of files in the dataframe is the same. @rragundez I think I didn't author this.

Comment (rragundez) [reply tests/image/dataframe_iterator_test.py]: thanks @Vijayabhaskar96

**Comment by rragundez** [comment]
I've change the PR taking into account the comments @Dref360

**Pull request merged**